src/app/App.tsx

```diff
@@ -443,6 +468,48 @@ class App extends AsyncComponent<RouteComponentProps, AppState> {
 
         document.title = title;
     }
+
+    /**
+     * Refresh the token one minute before it expires.
+     */
+    private validateTokenPeriodically() {
+        this.clearTokenExpiryInterval();
+        const jwt = this._storageService.load<string>("dashboard-jwt");
+        const expiryTimestamp = this.getTokenExpiry(jwt);
+
+        this._tokenExpiryTimer = setInterval(async () => {
+            if (moment(expiryTimestamp).isBefore(moment())) {
+                this._authService.logout();
+                this.clearTokenExpiryInterval();
+            } else if (moment(expiryTimestamp).isBefore(moment().valueOf() + MILLI_SECONDS_IN_ONE_MINUTE)) {
```

Comment:
This one is also confusing...maybe:
```
now.isBetween(moment(expiryTimestamp).subtract(1, "minute"), moment(expiryTimestamp))
...
```
or do even better by having the `expiryTimestamp` already parsed to `moment` before the `setInterval` so that you work with `moment` and not `number`
```
const expiryTimestamp = this.getTokenExpiry(jwt);
const expiryDate = moment(expiryTimestamp);
const refreshTokenDate = expiryDate.subtract(1, "minute");
...
...
else if (now.isBetween(refreshTokenDate, expiryDate)) {
...
```
Looks much more readable right ?
Also removes the need for `MILLI_SECONDS_IN_ONE_MINUTE`